File: JavaScript/647. Palindromic Substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var countSubstrings = function(s) {
    var res = 0;
    var n = s.length;
    for(var i = 0; i<n; i++){
        var j = i;
        var k = i;
        while(j>=0 && k<n && s[j] == s[k]){
            j--;
            k++;
            res++;
        }
        j = i;
        k = i+1;
        while(j>=0 && k<n && s[j] == s[k]){
            j--;
            k++;
            res++;
        }
    }
    return res;
};

File: C++/647. Palindromic Substrings.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int res = 0;
        int n = s.size();
        for(int i = 0; i<n; i++){
            int j = i;
            int k = i;
            while(j>=0 && k<n && s[j] == s[k]){
                j--;
                k++;
                res++;
            }
            j = i;
            k = i+1;
            while(j>=0 && k<n && s[j] == s[k]){
                j--;
                k++;
                res++;
            }
        }
        return res;
    }
};
